framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1157,6 +1157,12 @@ public AnnotatedTypeMirror getAnnotatedTypeFromTypeTree(Tree tree) {
         return qualifierUpperBounds.getBoundQualifiers(type);
     }
 
+    /** Returns the set of qualifiers that are the upper bounds for a use of the type. */
+    public Set<AnnotationMirror> getTypeDeclarationBounds(TypeElement typeElement) {
+        return qualifierUpperBounds.getBoundQualifiers(
+                AnnotatedTypeMirror.createType(typeElement.asType(), this, true));
```

Comment:
It seems rather costly to create an `AnnotatedTypeMirror` here. `QualifierUpperBounds.getBoundQualifiers` then deconstructs it into an Element again very soon
https://github.com/typetools/checker-framework/blob/defaultyChanges/framework/src/main/java/org/checkerframework/framework/type/QualifierUpperBounds.java#L122

Maybe an `Element` version can be added there?